### PR TITLE
Improvements for building wheels

### PR DESCRIPTION
**Pull request by ksivaman**
# Description

Make build time improvements and bug fixes for publishing wheels.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Infra/Build change
- [ ] Code refractor

## Changes

- Add build for metapackage.
- Catch incorrect installations during import and provide logging messages.
- Improve scripts for wheel build.
- Bug fix for wheels introduced in #946 that needs xla include directory during jax extensions build.

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/__init__.py [hunk 1]
```diff
@@ -6,25 +6,54 @@
...
 
 def _load_library():
     """Load shared library with Transformer Engine C extensions"""
+    module_name = "transformer_engine_torch"
+
+    if is_package_installed(module_name):
+        assert is_package_installed("transformer_engine"), "Could not find `transformer-engine`."
+        assert is_package_installed(
+            "transformer_engine_cu12"
+        ), "Could not find `transformer-engine-cu12`."
```
Comment (timmoon10) [inline transformer_engine/pytorch/__init__.py:28]: Does this work for local builds? It seems `transformer_engine_cu12` is only installed in `build_wheels.sh`, which is called in the wheel Dockerfiles.

Comment (ksivaman) [reply transformer_engine/pytorch/__init__.py:28]: This check is only run when the framework level module `transformer_engine_torch` is found, which is not the case for local builds

**Review by timmoon10 — APPROVED** [review]
LGTM

**Pull request merged**